### DISCUSSION TITLE
Linear advance

### DISCUSSION THREAD

**Discussion by PossibleCause** [Jyers/Marlin#811] [Ideas]
Hi, is linear advance going to be included in the near future for the 4.2.7. Boards?

**Comment by smooreace** [comment]
Search is your friend!

**Comment by Reggi3** [comment]
The 4.2.7 board configs are almost identical to the 4.2.2 configs, iirc the only major change needed is from:
```#define MOTHERBOARD BOARD_CREALITY_V4```
To:
```#define MOTHERBOARD BOARD_CREALITY_V427```

The above change will make any jyers or E3v2 4.2.2 based config into a 4.2.7 config. This is also true for any other E3v2 based firmware/config combo out there, as far as marlin is concerned, the only differences it needs to know about are the pin changes, which are handled by that one define change above.

Things are slightly different for the other 4.2.x boards that come stock on the whole Ender 3 range now, as some of them have slightly different driver chips and do not have a dwin based LCD.

**Reply by PossibleCause** [reply]
As I understand the 4.2.2 board is not very handy for linear advance because of the TMC2208 drivers (Linear advance is only available in UART mode). The 4.2.7 boards have the TMC2225 drivers which are more advanced but you need to change the firmware to make it work together with some code in your slicer. That’s why I asked if linear advance will be included in the near future. Mayby as an advanced option?
